File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/asistencia/AttendanceRealtimeEvent.java
package pe.albrugroup.schedule_service.entity.response.asistencia;

import lombok.Builder;
import lombok.Getter;
import pe.albrugroup.schedule_service.entity.enums.EstadoAsistencia;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Builder
public class AttendanceRealtimeEvent {

    private String tipo;
    private String origen;
    private Long idEmpleado;
    private LocalDate fecha;
    private EstadoAsistencia estadoActual;
    private EstadoAsistencia estadoAnterior;
    private LocalDateTime desde;
    private Boolean tieneHorarioVigente;
    private Boolean laborableHoy;
    private Boolean esperadoHoy;
    private Boolean operativo;
    private Instant occurredAt;
}

File: frontend/src/app/shared/models/schedule/attendance-realtime-event.ts
export interface AttendanceRealtimeEvent {
  tipo: string;
  origen: string;
  idEmpleado: number;
  fecha: string;
  estadoActual?: string | null;
  estadoAnterior?: string | null;
  desde?: string | null;
  tieneHorarioVigente?: boolean | null;
  laborableHoy?: boolean | null;
  esperadoHoy?: boolean | null;
  operativo?: boolean | null;
  occurredAt?: string | null;
}
